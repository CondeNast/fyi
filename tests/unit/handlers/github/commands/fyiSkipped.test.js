const fyiSkippedEvent = require('../../../fixtures/fyi-skipped')
const models = require('../../../../../src/models')

let app
let github

describe('Arch Bot', () => {
  beforeEach(() => {
    ({app, github} = require('../../../setup.js'))
  })

  describe('FYI Skipped', () => {
    it('Closes Issue in Admin Repository', async () => {
      await app.receive(fyiSkippedEvent)
      expect(github.issues.removeLabel).toMatchSnapshot()
      expect(github.issues.addLabels).toMatchSnapshot()
      expect(github.issues.createComment).toMatchSnapshot()
      expect(github.issues.edit).toMatchSnapshot()
    })
  })

  afterAll(async () => {
    await models.sequelize.close()
  })
})
