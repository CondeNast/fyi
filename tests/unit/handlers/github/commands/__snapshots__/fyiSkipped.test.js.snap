// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arch Bot FYI Skipped Closes Issue in Admin Repository 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "name": "repo-created",
        "number": 291,
        "owner": "choosenearme",
        "repo": "fyis",
      },
    ],
  ],
}
`;

exports[`Arch Bot FYI Skipped Closes Issue in Admin Repository 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "labels": Array [
          "fyi-skipped",
        ],
        "number": 291,
        "owner": "choosenearme",
        "repo": "fyis",
      },
    ],
  ],
}
`;

exports[`Arch Bot FYI Skipped Closes Issue in Admin Repository 3`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "body": "@gautamarora skipped the request for FYI.",
        "number": 291,
        "owner": "choosenearme",
        "repo": "fyis",
      },
    ],
  ],
}
`;

exports[`Arch Bot FYI Skipped Closes Issue in Admin Repository 4`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "number": 291,
        "owner": "choosenearme",
        "repo": "fyis",
        "state": "closed",
      },
    ],
  ],
}
`;
